src/axom/quest/detail/shaping/shaping_helpers.hpp

```diff
@@ -13,21 +13,158 @@
 #define AXOM_QUEST_SHAPING_HELPERS__HPP_
 
 #include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/primal.hpp"
 
-#if !defined(AXOM_USE_MFEM)
-  #error Sampling-shaping functionality requires Axom to be configured with MFEM and the AXOM_ENABLE_MFEM_SIDRE_DATACOLLECTION option
+#if defined(AXOM_USE_MFEM)
+  #include "mfem.hpp"
 #endif
 
-#include "mfem.hpp"
-
 namespace axom
 {
+
+template <typename Signature, size_t MaxSize = 16>
+class function;
```

Comment:
Here's the `axom::function` -- which is intended to be a device compatible drop-in for `std::function`.

I'm planning to generalize this and move it to axom's core component in a follow-up PR since I need it for the point projector lambdas in the sample-based shaping.